docs/api_auth.md

```diff
@@ -42,6 +42,37 @@ sequenceDiagram
     Relay->>Extension: [{id:, address:, etc.}, {id:, address:, etc.}]
 ```
 
+## Firefox OAuth Token Authentication and Accept Terms of Service
+
+Similarly to the add-on, Firefox uses the [the FXA OAuth service][fxa-oauth] /oauth/token endpoint with `scope: ["https://identity.mozilla.com/apps/relay"]` to get the scoped access token that expires every 24 hours (see which calls [`getOAuthToken`][searchfox-getoauthtoken] [`accessTokenWithSessionToken`][searchfox-accesstokenwithsessiontoken]).
+
+After successful OAuth flow with FXA, Firefox uses this token to authenticate all requests to Relay by including an `Authorization: Bearer {fxa-access-token}` header in all API requests. Since the token was generated from FXA and not in Relay, Firefox must hit Relay's /api/v1/terms-accepted-user endpoint to ensure that user has accepted the Terms of Service and for a new user and profile to be created on Relay.
+
+```mermaid
+sequenceDiagram
+    participant Firefox
+    participant FXA
+    participant Relay
+
+rect rgba(34, 0, 51, .09)
+    note right of Firefox: OAuth2 Flow
+    Firefox->>FXA: POST /oauth/token/
+    FXA->>Firefox: JSON: {Relay access token valid for 24 hrs}
+    end
```

Comment:
suggestion (non-blocking):
Instead of `note`, We could use [`loop`](https://mermaid.js.org/syntax/sequenceDiagram.html#loops) or [`alt`](https://mermaid.js.org/syntax/sequenceDiagram.html#alt) syntax here to make mermaid draw this note as a box that "wraps around" the relevant steps:
```suggestion
    alt OAuth2 Flow
        Firefox->>FXA: POST /oauth/token/
        FXA->>Firefox: JSON: {Relay access token valid for 24 hrs}
    end
```

Reply:
Thanks! I agree that encapsulating/highlighting the different parts of the flow will help better visualize and understand the flow. I looked at the documents on Mermaid and I think it will be better to accomplish this using [background highlighting](https://mermaid.js.org/syntax/sequenceDiagram.html#background-highlighting) since the [alt tag](https://mermaid.js.org/syntax/sequenceDiagram.html#alt) is used for alternative paths.

Reply:
Completed: c4b8e2c